starlite/datastructures.py

```diff
@@ -368,3 +368,14 @@ async def close(self) -> None:
         for _, value in self.multi_items():
             if isinstance(value, UploadFile):
                 await value.close()
+
+
+class AdditionalResponse(BaseModel):
+    """Container type of additional responses."""
+
+    model: Type[BaseModel]
+    """A model that describes the content of the response"""
+    generate_examples: bool = True
+    """Generate examples for the response content"""
+    description: str = "Additional response"
+    """A description of the response"""
```

Comment:
I'm not sure this is the right place for this class, please advise if you think we should move it

Reply:
Seems reasonable enough to me.

Reply:
I would suggest to rename this from `AdditionalResponse` to `ResponseSpec` or something along these lines. I would also move it out of `starlite/datastructures` and place it under `starlite/openapi/datastructures` - unless there is extra logic to add there, in which case its fine to keep in this loc.

Reply:
Addressed in https://github.com/starlite-api/starlite/pull/478/commits/dc2b02e52214f4ef2be388ce4d5f114615bea6bb